src/App.tsx

```diff
@@ -14,11 +14,110 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  goods: string[],
+  goodsVisible: boolean,
+  isSorted: boolean,
+  isSortLength: boolean,
+}
 
-export default App;
+export class App extends React.Component<{}, State> {
+  state: State = {
+    goods: goodsFromServer,
+    goodsVisible: false,
+    isSorted: false,
+    isSortLength: false,
+  };
+
+  showGoods = () => (this.setState(state => ({
```

Comment:
Use more appropriate names.
```suggestion
  showOrHideGoods = () => (this.setState(state => ({
```
```suggestion
  toggleGoods = () => (this.setState(state => ({
```